### ISSUE TITLE
glCreateShader call fails with v3.0.2

### ISSUE THREAD

**Issue by 9prady9** [cginternals/glbinding#278]
I am facing an issue while porting [forge](https://github.com/arrayfire/forge/) visulization library from 2.* version to 3.0.2. I was able to fix the compilation issues, at least I thought so until I saw the runtime issues. 

I initially thought it might be due to `GetProcAddress` I am passing, but I think I am doing it correctly - since the context relevant is current when I call `Binding::initialize`.

Please have a look at the changes I made to forge at https://github.com/arrayfire/forge/pull/184 The code compiles fine but the **glCreateShader** calls return 0 always at runtime. Can you please suggest what I might be missing in the migration from 2.* to 3.0.

Thank you,
Pradeep.

**Comment by scheibel** [comment]
Regarding your PR and its diff, I don't see any issue.

However, you perform calls to both `glbinding::Binding::useCurrentContext` and `glbinding::Binding::initialize`. This is not necessary as (1) the `initialize` takes a parameter that indicates immediate use after initialization (which you seem to want) and (2) the initial `useCurrentContext` performs an internal initialization as if it was called without function pointer resolve callback.

In conclusion, this double initialization may be a cause for an error, although a preliminary source code check didn't raise anything suspicious.

**Comment by 9prady9** [comment]
@scheibel Thanks for the comments, Will remove the redundant call and let you know how that goes.

**Comment by 9prady9** [comment]
@scheibel It works but there is strange statement printed onto the console. The rendering seems to be fine, but it thinks GLFW is not initialized.

```
./build/examples/cuda/surface 
The GLFW library is not initialized
```
Here is the diff of change I did after the PR.
```diff
diff --git a/src/backend/opengl/window_impl.cpp b/src/backend/opengl/window_impl.cpp
index fca909d..3f12cd5 100644
--- a/src/backend/opengl/window_impl.cpp
+++ b/src/backend/opengl/window_impl.cpp
@@ -129,9 +129,8 @@ window_impl::window_impl(int pWidth, int pHeight, const char* pTitle,
         mWindow = new wtk::Widget(pWidth, pHeight, pTitle, nullptr, invisible);
     }
     /* Set context (before glewInit()) */
-    MakeContextCurrent(this);
 
-    glbinding::Binding::useCurrentContext();
+    mWindow->makeContextCurrent();
     glbinding::Binding::initialize(mWindow->getProcAddr(), false);
 
     mCxt = mWindow->getGLContextHandle();
```
The windows show correct rendering and closes fine when I press escape as it should.

I don't see this problem with SDL2. Hence, I don't believe this is issue with glbinding anymore. It either has something to do with glfw or forge itself.

Thanks for you help!
